### PR TITLE
Dynamic file-naming to retain copy of viz timeseries data

### PR DESCRIPTION
**Pull request by lindsayplatt**
Added a method to allow us to save the current visualization's timeseries data, while also not having to change the target names & Vue code for each build. This fixes one of the things mentioned in #7.

### CODE REVIEW

**Comment by lindsayplatt** [comment]
@ceenell note that I have rebuilt the historic components and the data part for the past year. I will make a separate PR with those indicator files, so that you can pick up from that spot.

File: README.md [hunk 1]
```diff
@@ -34,11 +34,14 @@ In the end the following targets should have been built successfully (because th
...
+Follow the two steps below in order to retain a copy of the current visualization's timeseries data, so that we can access it in the future without a data pipeline rebuild. 
+
+1. Authenticate to AWS using the Dev VPC. Use `saml2aws login` in your regular command line, and then make sure you choose the `gs-chs-wma-dev` account when prompted. If you are still authenticated to the Prod VPC, try running `saml2aws login --force` in order to force a new login.
+1. Then run `scmake('3_visualize/out/gw-conditions-peaks-timeseries-s3copy.ind')` to push a copy of the `3_visualize/out/gw-conditions-peaks-timeseries.csv` file to the `vizlab-data` bucket on the Dev VPC. The file will be automatically given a new name based on the `viz_start_date` and `viz_end_date` target values, using this pattern: `gw-conditions/viz-previous-peak-data/gw-conditions-peaks-timeseries-[viz_start_date]_[viz_end_date].csv`.
```
Comment (ceenell) [inline README.md:44]: good tip

**Review by ceenell — APPROVED** [review]
This looks good - I did not run the pipeline but do not see any issues with the changes made. 

Wondering if `3_visualize/out/gw-conditions-site-coords.csv` needs to be adapted for rebuild also. If I remember correctly, that file reflects all the sites with a historic record that fit our criteria, and not whether or not there were active sites for a given time period. Is that accureate?

**Comment by lindsayplatt** [comment]
> Wondering if `3_visualize/out/gw-conditions-site-coords.csv` needs to be adapted for rebuild also.

You're right that this contains only sites that appear in the historic record. That is how we limit our initial download though, the site must appear in our historic record in order to be downloaded (so that we can assign a quantile). It might contain extra sites per build. How would the Vue handle a site that appeared in this file but didn't have a matching groundwater level? I can add a step to filter this site list to only those with data during this time period if necessary. 

I don't think I'd want to apply the same approach we are using here to retain a copy of the data, though. Rebuilding site coordinates is a fairly inexpensive step, so I don't think we would need to retain a copy per time period like we are doing for the actual groundwater level timeseries data.

**Comment by ceenell** [comment]
> I don't think I'd want to apply the same approach we are using here to retain a copy of the data, though. Rebuilding site coordinates is a fairly inexpensive step, so I don't think we would need to retain a copy per time period like we are doing for the actual groundwater level timeseries data.

Agree, but if we set this up so there's a menu selection for different time periods that could cause some issues with mismatching between the timeseries and site data. Not clear how significant this would be.

In vue it's set up so it is throwing an error for sites without data, but if there's partial data with intermixed NA's it comes in and out with the data. In our initial version this was necessary because a number of sites didn't have data on the first date but we still wanted them for later in the year.

I think we should add a filtering step in vue that takes the full site list and active sites for a given time frame.

**Comment by lindsayplatt** [comment]
> I think we should add a filtering step in vue that takes the full site list and active sites for a given time frame.

So no need to retain an additional copy with this approach, right? Just want to make sure I am interpreting correctly.

**Comment by ceenell** [comment]
> > I think we should add a filtering step in vue that takes the full site list and active sites for a given time frame.
> 
> So no need to retain an additional copy with this approach, right? Just want to make sure I am interpreting correctly.

yes correct, I'll make an issue for myself

**Pull request merged**